### PR TITLE
doc(ant,maven): add labels for description

### PR DESCRIPTION
**Pull request by vpa1977**
Ant and maven plugin descriptions are overriden by rockcraft. Add labels to simplify documentation override.

- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

**Comment by vpa1977** [comment]
Note: ruff shows the following error:
```
craft_parts/executor/environment.py:276:5: PLC0206 Extracting value from dictionary without calling `.items()`
    |
274 |       )
275 |   
276 |       for key in data:
    |  _____^
277 | |         if not skip or key not in skip:
278 | |             data[key] = _replace_attr(data[key], replacements)
    | |______________________________________________________________^ PLC0[20](https://github.com/canonical/craft-parts/actions/runs/12108844086/job/33757305803?pr=916#step:7:21)6
    |

Found 1 error.
make: *** [Makefile:91: test-ruff] Error 1
```
Probably due to mutating the dictionary while iterating through the keys.

File: docs/common/craft-parts/reference/plugins/ant_plugin.rst [hunk 1]
```diff
@@ -3,6 +3,7 @@
 Ant plugin
 ==========
 
+.. _craft_parts_ant_plugin_description_begin:
```
Comment (tigarmo) [inline docs/common/craft-parts/reference/plugins/ant_plugin.rst:6]: should we scope this further and only surround the "After a successful build..." plus the steps? This initial paragraph is still true for all *crafts right?

Maybe `_craft_parts_ant_plugin_post_build_begin/end`

**Comment by mr-cal** [comment]
> Note: ruff shows the following error:
> 
> ```
> craft_parts/executor/environment.py:276:5: PLC0206 Extracting value from dictionary without calling `.items()`
>     |
> 274 |       )
> 275 |   
> 276 |       for key in data:
>     |  _____^
> 277 | |         if not skip or key not in skip:
> 278 | |             data[key] = _replace_attr(data[key], replacements)
>     | |______________________________________________________________^ PLC0[20](https://github.com/canonical/craft-parts/actions/runs/12108844086/job/33757305803?pr=916#step:7:21)6
>     |
> 
> Found 1 error.
> make: *** [Makefile:91: test-ruff] Error 1
> ```
> 
> Probably due to mutating the dictionary while iterating through the keys.

Yep, I'll put up a PR to fix this unrelated failure whenever we come to an agreement in https://github.com/canonical/craft-application/pull/580

**Comment by mr-cal** [comment]
Nevermind, Tiago already cleaned it up in https://github.com/canonical/craft-parts/pull/922/files

**Review by lengau — APPROVED** [review]

**Review by tigarmo — APPROVED** [review]

File: docs/common/craft-parts/reference/plugins/ant_plugin.rst [hunk 2]
```diff
@@ -9,12 +9,14 @@ path to the latest JDK found in the build environment.
 
 After a successful build, this plugin will:
 
+.. _craft_parts_ant_plugin_post_build_begin:
 * Create ``bin/`` and ``jar/`` directories in ``$CRAFT_PART_INSTALL``.
```
Comment (tigarmo) [inline docs/common/craft-parts/reference/plugins/ant_plugin.rst:13]:
```suggestion

* Create ``bin/`` and ``jar/`` directories in ``$CRAFT_PART_INSTALL``.
```

File: docs/common/craft-parts/reference/plugins/maven_plugin.rst [hunk 3]
```diff
@@ -9,12 +9,14 @@ path to the latest JDK found in the build environment.
 
 After a successful build, this plugin will:
 
+.. _craft_parts_maven_plugin_post_build_begin:
 * Create ``bin/`` and ``jar/`` directories in ``$CRAFT_PART_INSTALL``.
```
Comment (tigarmo) [inline docs/common/craft-parts/reference/plugins/maven_plugin.rst:13]:
```suggestion

* Create ``bin/`` and ``jar/`` directories in ``$CRAFT_PART_INSTALL``.
```

**Pull request merged**